v2/kr/kr_holidays.go

```diff
@@ -0,0 +1,205 @@
+        Type:     cal.ObservancePublic,
+        Month:    time.May,
+        Day:      5,
+        Observed: weekendAlt,
+        Func:     cal.CalcDayOfMonth,
+    }
+
+    // BuddhasDay represents 부처님 오신 날 on the Lunar 8-April
+    BuddhasDay = &cal.Holiday{
+        Name:  "부처님 오신 날",
+        Type:  cal.ObservancePublic,
+        Month: time.April,
+        Day:   8,
+        Lunar: true,
+        Func:  CalcDayOfLunarMonth,
+    }
+
+    // MemorialDay represents 현충일 on the 6-June
+    MemorialDay = &cal.Holiday{
+        Name:  "현충일",
+        Type:  cal.ObservancePublic,
+        Month: time.June,
+        Day:   6,
+        Func:  cal.CalcDayOfMonth,
+    }
+
+    // LiberationDay represents 광복절 on the 15-August
+    LiberationDay = &cal.Holiday{
+        Name:  "광복절",
+        Type:  cal.ObservancePublic,
+        Month: time.August,
+        Day:   15,
+        Func:  cal.CalcDayOfMonth,
+    }
+
+    // BeforeChuseok represents 추석전날 before on the Lunar 15-August
+    BeforeChuseok = &cal.Holiday{
+        Name:  "추석전날",
+        Type:  cal.ObservancePublic,
+        Month: time.August,
+        Day:   15,
+        Lunar: true,
+        Func:  CalcBeforeDayOfLunarMonth,
+    }
+
+    // Chuseok represents 추석 on the Lunar 15-August
+    Chuseok = &cal.Holiday{
+        Name:     "추석",
+        Type:     cal.ObservancePublic,
+        Month:    time.August,
+        Day:      15,
+        Lunar:    true,
+        Observed: LunarNewYearChuseokAlt,
+        Func:     CalcDayOfLunarMonth,
+    }
+
+    // AfterChuseok represents 추석전날 after on the Lunar 15-August
+    AfterChuseok = &cal.Holiday{
+        Name:  "추석다음날",
+        Type:  cal.ObservancePublic,
+        Month: time.August,
+        Day:   15,
+        Lunar: true,
+        Func:  CalcAfterDayOfLunarMonth,
+    }
+
+    // FoundationDay represents 개천절 on the 3-October
+    FoundationDay = &cal.Holiday{
+        Name:  "개천절",
+        Type:  cal.ObservancePublic,
+        Month: time.October,
+        Day:   3,
+        Func:  cal.CalcDayOfMonth,
+    }
+
+    // HangeulDay represents 개천절 on the 9-October
+    HangeulDay = &cal.Holiday{
+        Name:  "한글날",
+        Type:  cal.ObservancePublic,
+        Month: time.October,
+        Day:   9,
+        Func:  cal.CalcDayOfMonth,
+    }
+
+    // ChristmasDay represents 성탄절 on the 25-Dec
+    ChristmasDay = aa.ChristmasDay.Clone(&cal.Holiday{Name: "성탄절", Type: cal.ObservancePublic})
+
+    // Holidays provides a list of the standard national holidays
+    Holidays = []*cal.Holiday{
+        NewYear,
+        BeforeLunarNewYear,
+        LunarNewYear,
+        AfterLunarNewYear,
+        IndependenceMovementDay,
+        BuddhasDay,
+        ChildrensDay,
+        MemorialDay,
+        LiberationDay,
+        BeforeChuseok,
+        Chuseok,
+        AfterChuseok,
+        FoundationDay,
+        HangeulDay,
+        ChristmasDay,
+    }
+)
+
+// CalcDayOfLunarMonth calculates the occurrence of a holiday that is always a
```

Comment:
Do you think these would be used by other lunar calendars? Maybe these functions along with data.go and lunar.go should be moved to something like 'aa/lunar' where it can be shared?

Reply:
korean, chinese and japanese lunar calendars are different.
each lunar calendar should implements on they requirements.

I don't think it needs to be shared.